### PR TITLE
Double symmetry

### LINKED ISSUE #90: mirror/symentry only works at horizontal or vertical at one time

### LINKED ISSUE #105: We need to have both horizontal and vertical symmetry at the same time
Honestly, I really need this feature

### PR DESCRIPTION
**Pull request by Zughy**
Closes #90, closes #105 .

I'm no C++ dev, just an amateur, so I hope the code is... well, decent. I also avoided opening Aseprite code, as I didn't want to be influenced by how David solved this problem (and because it's proprietary, so I really shouldn't copy-paste).

Marking it as a draft as I'd like a @zJudGenie check first, who so kindly helped me through the whole process.
### To-do
- [x] code cleaning

Result:

https://user-images.githubusercontent.com/63455151/128437117-7169175a-b5cb-4108-a56e-e73acb05aa7a.mp4

### CODE REVIEW

**Comment by maujin111** [comment]
I will review/check this tomorrow night

File: src/app/ui/context_bar.cpp [hunk 1]
```diff
@@ -1284,7 +1284,8 @@ class ContextBar::SymmetryField : public ButtonSet {
 
     DocumentPreferences& docPref = Preferences::instance().document(doc);
 
-    setSelectedItem((int)docPref.symmetry.mode());
+    at(0)->setSelected((int)docPref.symmetry.mode() == 1 || (int)docPref.symmetry.mode() == 3 ? true : false);
+    at(1)->setSelected((int)docPref.symmetry.mode() >= 2 ? true : false);
```
Comment (zJudGenie) [inline src/app/ui/context_bar.cpp:1288]: There is no need to use the ternary operator, also I think that is better to use the SymmetryMode enum and not the numerical value, to improve code readability.
```suggestion
    app::gen::SymmetryMode symmetryMode = docPref.symmetry.mode()
    at(0)->setSelected(symmetryMode == app::gen::SymmetryMode::HORIZONTAL
                         || symmetryMode == app::gen::SymmetryMode::BOTH);
    at(1)->setSelected(symmetryMode == app::gen::SymmetryMode::VERTICAL ||
                         || symmetryMode == app::gen::SymmetryMode::BOTH);
```

**Review by maujin111 — APPROVED** [review]

**Comment by PerryHugh** [comment]
Why the revert @maujin111?

**Comment by PerryHugh** [comment]
PS: @Zughy this is freaking sweet, thank you :)

**Comment by maujin111** [comment]
@PerryHugh never mind, must be a mistake,

**Comment by PerryHugh** [comment]
@maujin111 Could you unrevert it?

**Comment by maujin111** [comment]
The revert request o or the pull request? Anyhow, i'can't., just close the pr.

**Comment by PerryHugh** [comment]
@maujin111 It says:
_maujin111 added a commit that referenced this pull request 2 days ago
@maujin111
Revert "Double symmetry (#132)"_ 

Can you revert this commit, so the feature is implemented?

**Comment by maujin111** [comment]
@PerryHugh  the feature is already implement

**Pull request merged**